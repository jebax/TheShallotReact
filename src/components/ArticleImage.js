import React from 'react'
import PropTypes from 'prop-types'

const ArticleImage = ({ imageUrl }) => {
  return (
    <figure
      className='imageWrapper'
    >
      <img
        alt=''
        className='articleImage'
        src={imageUrl}
      />
    </figure>
  )
}

ArticleImage.propTypes = {
  imageUrl: PropTypes.string.isRequired
}

export default ArticleImage
