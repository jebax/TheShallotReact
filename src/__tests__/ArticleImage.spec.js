import React from 'react'
import { shallow, mount } from 'enzyme'
import ArticleImage from '../components/ArticleImage'

describe('<ArticleImage />', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(<ArticleImage imageUrl={'TestImage'} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders an image wrapper', () => {
    expect(wrapper.find('.image-wrapper').length).toEqual(1)
  })

  it('renders an image', () => {
    expect(wrapper.find('.article-image').length).toEqual(1)
  })

  it('renders its image based on its image url', () => {
    const image = wrapper.find('.article-image')

    expect(image.props().src).toEqual('TestImage')
  })
})
