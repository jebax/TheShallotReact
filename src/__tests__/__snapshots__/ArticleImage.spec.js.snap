// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleImage /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleImage
    imageUrl="TestImage"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt=""
        className="articleImage"
        src="TestImage"
      />,
      "className": "imageWrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "",
        "className": "articleImage",
        "src": "TestImage",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "figure",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt=""
          className="articleImage"
          src="TestImage"
        />,
        "className": "imageWrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "articleImage",
          "src": "TestImage",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "figure",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
